### DISCUSSION TITLE
Can i run whisper on arm32 cpu

### DISCUSSION THREAD

**Discussion by yetgintarikk** [openai/whisper#1967] [Q&A]
I am using stm32mp157c-dk2 board which has Arm®-based dual Cortex®-A7 32 bits cpu . I want to run a python program uses whisper. Can my board run it?

**Comment by phineas-pta** [comment]
whisper rely on pytorch which does support arm so u can try

**Reply by yetgintarikk** [reply]
Thanks for reply,
I am having trouble installing numpy. Are you sure pytorch support ARMv7 architecture.

**Reply by phineas-pta** [reply]
they have wheel files for arm architecture, but i dont have your hardware to verify

**Comment by Mattral** [comment]
You might wanna try this one
https://github.com/ggerganov/whisper.cpp

or
https://www.youtube.com/watch?v=ibDY1PKRQZU
 Voice Recognition on Arm Cortex-M4

Plus you may have problem getting different results on two different CPU architectures Itel and ARM

**Reply by gyllila** [reply]
Yes, it also runs on CPU. If you just want to quickly transcribe a file, you can try out my easy_whisper, it detects automatically if you have CUDA to run Whisper or have to use CPU. Here the GitHub link:
https://github.com/gyllila/easy_whisper

**Reply by yetgintarikk** [reply]
Thanks for reply,
Is it english only? I need to transcribe on another language.

**Reply by gyllila** [reply]
You can find here which languages are supported by OpenAI Whisper (and thus easy_whisper):
https://github.com/openai/whisper#available-models-and-languages

**Reply by phineas-pta** [reply]
@gyllila OP asked for arm cpu not general x86, your answer isn't related

**Reply by gyllila** [reply]
I don’t really know the difference between arm and x86, but given the answer of Mattral I thought yetgintarikk can use OpenAI Whisper, thus also my easy_whisper, which just adds a (double) friendly user interface to OpenAI Whisper and makes transcription of longer audio faster by splitting the audio into sentences, which makes the context shorter for higher speed but still long enough for a good quality. Using only base.en, I can transcribe my English audiobooks almost free of error, I’m really impressed by the quality of OpenAI Whisper models.

**Reply by phineas-pta** [reply]
mattral's answer is not openai whisper, it's a c++ version to run on different hardware platforms, yours is just a wrapper around existing python code

**Reply by gyllila** [reply]
Thank you for the correction. If yetgintarikk can’t use OpenAI Whisper on his computer, then he can’t use my easy_whisper either. I’m glad he didn’t install my library or OpenAI’s python library in vain.
